apis/core/v1alpha2/core_types.go

```diff
@@ -49,6 +50,10 @@ type WorkloadDefinitionSpec struct {
 
 	// ChildResourceKinds are the list of GVK of the child resources this workload generates
 	ChildResourceKinds []ChildResourceKind `json:"childResourceKinds,omitempty"`
+
+	// Extension is used for extension needs by OAM platform builders
+	// +optional
+	Extension *unstructured.Unstructured `json:"extension,omitempty"`
```

Comment:
I think one Unstructured as extension might not be enough. We should design it for more future extension needs.

How about change it to:
```
Extensions map[string]*Unstructured
```

So we could have:

```
extensions:
  template:
    ...
```

Reply:
Current struct can also represent. The underlying go structure is map[string]interface{} which means can represent any object.

```
extensions:
  template:
    ...
```